engine/gamesys/src/gamesys/components/comp_gui.cpp

```diff
@@ -2116,9 +2122,24 @@ namespace dmGameSystem
 
     dmGameObject::PropertyResult CompGuiSetProperty(const dmGameObject::ComponentSetPropertyParams& params) {
         GuiComponent* gui_component = (GuiComponent*)*params.m_UserData;
-        if (params.m_PropertyId == PROP_MATERIAL) {
+        dmhash_t set_property = params.m_PropertyId;
+        if (set_property == PROP_MATERIAL) {
             return SetResourceProperty(dmGameObject::GetFactory(params.m_Instance), params.m_Value, MATERIAL_EXT_HASH, (void**)&gui_component->m_Material);
         }
+        else if (set_property == PROP_FONT) {
+            dmRender::HFontMap font;
+            memset(&font, 0, sizeof(dmRender::HFontMap));
```

Comment:
We generally don't do defensive programming.
Why is this necessary here?

You should check the result from the SetResourceProperty